com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/ProtocolServer.java

```diff
@@ -74,20 +77,36 @@ public void sendResponse(Messages.Response response) {
     @Override
     protected void dispatchRequest(Messages.Request request) {
         usageDataSession.recordRequest(request);
-        this.debugAdapter.dispatchRequest(request).whenComplete((response, ex) -> {
+        this.debugAdapter.dispatchRequest(request).thenAccept((response) -> {
             if (response != null) {
                 sendResponse(response);
-            } else if (ex != null) {
-                logger.log(Level.SEVERE, String.format("DebugSession dispatch exception: %s", ex.toString()), ex);
+            } else {
+                logger.log(Level.SEVERE, "The request dispatcher should not return null response.");
```

Comment:
Please just throw an exception with some message and the exceptionally will catch that.